homeassistant/components/media_player/sonos.py

```diff
@@ -62,6 +62,11 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
     if discovery_info:
         player = soco.SoCo(discovery_info)
+
+        # if device allready exists by config
+        if player.uid in [device.player_uid for device in DEVICES]:
```

Comment:
I can't use the test only on EntityComponent while sonos would add this also to DEVICES. But the `unique_id` is a greate idee.